compiler/AST/primitive.cpp

```diff
@@ -875,6 +874,10 @@ initPrimitive() {
 
   prim_def(PRIM_NEEDS_AUTO_DESTROY, "needs auto destroy", returnInfoBool, false, false);
   prim_def(PRIM_AUTO_DESTROY_RUNTIME_TYPE, "auto destroy runtime type", returnInfoVoid, false, false);
+
+  // Corresponds to LLVM's invariant start
+  // takes in a pointer/reference argument that is the invariant thing
+  prim_def(PRIM_INVARIANT_START, "invariant start", returnInfoVoid, false, false);
```

Comment:
I must be missing something obvious, but where are you marking the privatized load with this primitive?

Reply:
I'm not. This PR includes the primitive as something we can experiment with. Initially I was using it but then I figured out that the existing TBAA rules covered it (at least in the case we're working on here) and the LLVM LICM code doesn't work with llvm.invariant.start on a global variable.

However I'm expecting that this primitive will be useful for other LLVM LICM performance questions - I'm hoping to use it to try hoisting array headers, for example. But it's also not strictly necessary in this PR (or on master at all). (I tend to prefer to put such things on master so they are not lost / redone).